src/WebJobs.Script.WebHost/Metrics/LinuxContainerLegionMetricsPublisher.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO.Abstractions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Platform.Metrics.LinuxConsumption;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Metrics
+{
+    public class LinuxContainerLegionMetricsPublisher : IMetricsPublisher
+    {
+        private readonly ILinuxConsumptionMetricsTracker _metricsTracker;
+        private readonly LegionMetricsFileManager _metricsFileManager;
+        private readonly TimeSpan _memorySnapshotInterval = TimeSpan.FromMilliseconds(1000);
```

Comment:
The intervals for memory and publishing are the same as in LinuxContainerMetricsPublisher, The main change is how this activity is published. Here we publish to the ILinuxConsumptionMetricsTracker component directly, instead of buffering activities and flushing out via http as LinuxContainerMetricsPublisher does.